### PR TITLE
ARROW-4473: [Website] Support test site deployment (take 2)

### PR DESCRIPTION
**Pull request by nealrichardson**
See https://lists.apache.org/thread.html/73e8a7d2ee667e83371e8bb99c13338144c8af5444f55918994713be@%3Cdev.arrow.apache.org%3E

### CODE REVIEW

**Comment by nealrichardson** [comment]
@fsaintjacques PTAL. I added the `--porcelain` check as you requested on #8. As for the `perl` vs. `sed` comment there, at some previous point doing this on a different site, I had trouble with `sed` and using `perl` resolved it. I don't recall the details, but it seems moot since this script is designed to run on Travis, and Travis has `perl` just fine.

**Comment by fsaintjacques** [comment]
Script looks good to me, I only noted the suggested the perl change since the script was also using sed a bit earlier.

File: build-and-deploy.sh [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    perl -pe 's@^baseurl.*@baseurl: '"${BASE_URL}"'@' -i _config.yml
+
+    # Build
+    gem install jekyll bundler
+    bundle install
+    JEKYLL_ENV=production bundle exec jekyll build
+
+    # Publish
+    git clone -b ${TARGET_BRANCH} https://${GITHUB_PAT}@github.com/$TRAVIS_REPO_SLUG.git OUTPUT
+    rsync -r build/ OUTPUT/
```
Comment (kou) [inline build-and-deploy.sh]: `rsync -a --delete --exclude '/.git/' --exclude '/docs/' build/ OUTPUT/` is better to remove needless files.
(We may need more `--exclude`.)

**Comment by kou** [comment]
I think that we always use UTC but it'll break published posts' URL..

**Comment by nealrichardson** [comment]
I don't follow the time zone concern. Are you thinking that blog post URLs will change depending on the timezone of the server where the site is built? (If so, all the more reason to build and publish from CI so that's not dependent on who is updating the site.) Do you have evidence that URLs are being broken/changed?

**Comment by kou** [comment]
Sorry. This is not a problem for us. Timezone is already specified in `_config.yml`: https://github.com/apache/arrow-site/blob/master/_config.yml#L21

**Comment by nealrichardson** [comment]
Thanks for the suggestions for simplifying the script, @kou. I did them and the test build seemed to succeed: see https://enpiar.com/arrow-site/ for what was deployed. 

You may want to review the diff on the gh-pages branch there for what your proposed `rsync` command made. I think it's right but it deleted a lot of files--there seemed to be a copy of the website in a `_site` directory, and then there were some stale/misdated copies of blog posts that got removed too. https://github.com/nealrichardson/arrow-site/commit/b4431a6f2f2e2aa0bcec24e1f9b4b7130f9bc210 is huge (> 2M lines deleted) so it's probably easier to review starting around here in the Travis log: https://travis-ci.org/nealrichardson/arrow-site/builds/570970342#L6417

I spot-checked some of the deleted blog posts and there are copies of them with a different date (e.g. the 0.3.0 release post is at https://enpiar.com/arrow-site/blog/2017/05/08/0.3-release/, while in production there is the post at 2017/05/08 and then a (somewhat broken) copy at 05/07 (http://arrow.apache.org/blog/2017/05/07/0.3-release/), which seems to be what `rsync` removed.

**Comment by nealrichardson** [comment]
I suspect those duplicate (stale) blog posts are from before the timezone was hardcoded (https://github.com/apache/arrow-site/commit/aa19f5d845e2be843af649e340a7821ada7d9424). All but one are in 2017.

**Comment by kou** [comment]
I've created #10 to remove needless `_site/` directory.
We merge it then we can confirm diff for this change.

**Comment by nealrichardson** [comment]
👍 fine, you can have the -2,000,000 line diff :P 

FTR http://arrow.apache.org/_site/ does exist and shows a version of 0.12. https://github.com/apache/arrow-site/commit/99f824603cc862d3cef2eb6bad5a89aadd929951 is the commit that created that directory and the only one that has touched it, so definitely looks like an accident.

**Comment by kou** [comment]
#10 has been merged.
Could you rebase your fork?

**Comment by nealrichardson** [comment]
Here's the diff: https://github.com/nealrichardson/arrow-site/commit/a0d9803c06df8099fb916f24fff579edddf4fefe

Note that many of the line changes are the insertion of the relative path on my `gh-pages` branch. Here's the commit summary that names all of the deleted files: https://travis-ci.org/nealrichardson/arrow-site/jobs/570970343#L309

**Comment by nealrichardson** [comment]
IDK why that file mode would change, but it sounds like a good idea. No reason for a .png to be 755 is there?

/events/ looks like it hasn't been touched since 2016.

**Comment by nealrichardson** [comment]
@kou @wesm are we good here?

**Comment by kou** [comment]
It seems that `img/arrow.png`'s mode has been changed by https://github.com/apache/arrow-site/commit/dea4fb2e708151fc3f9ef07f44f6520770dc283b#diff-6ad51cc5e7d889c484fcb3597709f7d7 .
Could you restore mode of `img/arrow.png` to `0644`?
Then we can merge this.

**Comment by kou** [comment]
If we want to maintain `/events/` again, we should migrate the source of `/events/` to Jekyll based one.

**Comment by nealrichardson** [comment]
chmod done: https://github.com/apache/arrow-site/pull/9/commits/10bfe25e66765c0d6bc4e052cb236b20aabd5c5b

PTAL

**Comment by kou** [comment]
+1

**Pull request merged**